core/ledger/kvledger/delete_partial_ledgers_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package kvledger
+
+import (
+	"testing"
+
+	"github.com/hyperledger/fabric/core/ledger/kvledger/msgs"
+	"github.com/hyperledger/fabric/core/ledger/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeleteUnderDeletionLedger(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	conf.HistoryDBConfig.Enabled = true
+	defer cleanup()
+
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	defer provider.Close()
+
+	// Set up a ledger and mark it as "under deletion."  This emulates the
+	// state when a peer was in the process of unjoining a channel and the
+	// unjoin operation crashed mid delete.
+	ledgerID := constructTestLedger(t, provider, 0)
+	verifyLedgerIDExists(t, provider, ledgerID, msgs.Status_ACTIVE)
+
+	// doom the newly created ledger
+	require.NoError(t, provider.idStore.updateLedgerStatus(ledgerID, msgs.Status_UNDER_DELETION))
+	verifyLedgerIDExists(t, provider, ledgerID, msgs.Status_UNDER_DELETION)
+
+	// explicitly call the delete on the provider.
+	provider.deletePartialLedgers()
+
+	// goodbye, ledger.
+	verifyLedgerDoesNotExist(t, provider, ledgerID)
+}
```

Comment:
Though, the extra test does not harm and in this particular case it's very small enough addition that could easily be ignored. However, in general we should prefer to be thoughtful of the purpose of an additional test. 
- It's hard to think of a bug/issue which would make this test pass while the fail the next one. 
- Neither does this test tests a separate section of the code.